Content.Server/GameObjects/Components/Interactable/WelderComponent.cs

```diff
@@ -123,21 +128,15 @@ bool ExtraCheck()
             {
                 var extraCheck = doAfterCheck?.Invoke() ?? true;
 
-                return extraCheck && CanWeld(fuelConsumed);
+                return extraCheck && CanWeld(fuelConsumed, user);
             }
 
             return await base.UseTool(user, target, doAfterDelay, toolQualityNeeded, ExtraCheck) && TryWeld(fuelConsumed, user);
         }
 
         private bool TryWeld(float value, IEntity? user = null, bool silent = false)
         {
-            if (!WelderLit)
-            {
-                if(!silent) _notifyManager.PopupMessage(Owner, user, Loc.GetString("The welder is turned off!"));
-                return false;
-            }
-
-            if (!CanWeld(value))
+            if (!CanWeld(value, user))
```

Comment:
This if is suspect now.

imagine if you use a welder that's turned off here, you'll get both messages:

"The welder is turned off!"
"The welder does not have enough fuel for that!"

Reply:
When transferring from fuel tank for example it says it's off and that it has also transferred.

IMO I'd leave the notification out of CanWeld and just let the caller handle it which should fix it.